modules/javafx.base/src/main/java/javafx/beans/Observable.java

```diff
@@ -93,4 +94,23 @@ public interface Observable {
      */
     void removeListener(InvalidationListener listener);
 
+    /**
+     * Creates a {@link Subscription} on this value which calls the given
+     * {@code runnable} whenever it becomes invalid.
```

Comment:
I think the documentation for these new methods should go into more detail and clearly explain the semantics. These are fundamental concepts, and I think it's a good idea to err on the side of over-explaining. Here's a suggestion:

```
Creates a {@code Subscription} for invalidation events of this {@code Observable}.
<p>
The specified {@code runnable} will be invoked when the {@code Observable} transitions to the
<em>invalid</em> state. This happens when the observed value is changed, usually by calling
the {@link ObservableValue#setValue} method. Note that the specified {@code runnable} will not
be invoked for subsequent value changes unless the {@code Observable} has transitioned to the
<em>valid</em> state again, usually by invoking the {@link ObservableValue#getValue} method.
<p>
An {@code Observable} is automatically validated on every change when a {@link ChangeListener}
or a {@link Consumer} or {@link BiConsumer} subscription is added.
```

Reply:
Sure, that's a good suggestion, I may have been overly brief to avoid repeating what is known for invalidation listeners, but as this method does not mention invalidation listeners (and only uses them internally for now) a full explanation would be good to add.

Reply:
Another option would be to refer to the documentation for `InvalidationListener` or `Observable.addListener(InvalidationListener)`, both of which could be improved a bit in a separate PR.

Reply:
@mstr2 I updated the docs a bit, although I did not add your suggestion.  I looked at the full class, and I think the class docs for `Observable` already describes how invalidations work.  The docs for `addListener` don't repeat this either:

```
 * Implementations of this class should strive to generate as few events as
 * possible to avoid wasting too much time in event handlers. Implementations in
 * this library mark themselves as invalid when the first invalidation event
 * occurs. They do not generate anymore invalidation events until their value is
 * recomputed and valid again.
```

What I did do is add some more description similar to what the `addListener` methods have.